src/properties/forest.c

```diff
@@ -0,0 +1,216 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_structural.h"
+
+#include "igraph_adjlist.h"
+#include "igraph_constructors.h"
+#include "igraph_interface.h"
+#include "igraph_stack.h"
+
+/* count the number of vertices reachable from the root */
+static  igraph_error_t igraph_i_is_forest_visitor(igraph_integer_t root,igraph_vector_t *visited, const igraph_adjlist_t *al, igraph_integer_t *visited_count, igraph_bool_t *res, igraph_neimode_t mode){
```

Comment:
Please wrap long lines at around 80 chars or so to make it easier to read on narrow screens or for people who use two code panels side-by-side.